cmd/json.rb

```diff
@@ -142,6 +143,9 @@ def download_bottle(hash, tag)
     resource.version hash["pkg_version"]
     resource.downloader.resolved_basename = bottle_filename
 
+    # Remember that this formula name should be mapped to the newly downloaded bottle
+    Formulary.register_json_bottle hash["name"], resource.downloader.cached_location
```

Comment:
If https://github.com/Homebrew/brew/pull/11534 is merged: (also remove the entire `Formulary` monkey patch above)

```suggestion
    Formulary.map hash["name"], to: resource.downloader.cached_location
```